prusti-specs/src/specifications/preparser.rs

```diff
@@ -51,7 +51,11 @@ pub fn parse_prusti_assert_pledge(tokens: TokenStream) -> syn::Result<(TokenStre
 }
 
 pub fn parse_ghost_constraint(tokens: TokenStream) -> syn::Result<GhostConstraint> {
-    syn::parse2(tokens)
+    syn::parse2(tokens).map_err(|mut err| {
+        err.combine(syn::Error::new(err.span(),
+            "expected a trait bound `T: A + B` and specifications in brackets `[requires(...), ensures(...), pure, ...]`"));
```

Comment:
This is better, but still doesn't give an actual example of the syntax. I think the error message should contain the expected format `ghost_constraint(T: A + B + ..., [requires(...), ...])`